### ISSUE TITLE
Feature request: include current in amps (in addition to watts)

### ISSUE THREAD

**Issue by cmegalodon** [magico13/ha-emporia-vue#273]
I noticed that the integration provides 1MIN power in watts, but in the Emporia website it's possible to switch to amps.

My specific use case is monitoring power from a generator, and it's important to know the current on each leg (since the house is an unbalanced load).

**Comment by DabblerIOT** [comment]
can you describe your setup?  and service?

ty

**Comment by cmegalodon** [comment]
Yes, I'm in Canada, using 120V split phase power (+120 and -120, same as the U.S.). I've got a 12kW standby generator which is rated at 50 amps (120V/50A each leg). The Emporia integration provides total power in watts, but in reality the generator will trip its breaker if either leg exceeds 6kW.

In the Emporia app it's possible to "switch units" and see the amps on each leg (currently 12.8 amps and 20.5 amps on each leg, but about 3.8 kW). In the app it's reported as "Total Usage_A" and "Total Usage_B".

I'd like to set up Home Assistant to trigger an alert if either leg exceeds a certain current (say, 40 amps).

**Comment by DabblerIOT** [comment]
it shows you two amperages off main cts?  or you have individual circuit
cts on gen?

**Comment by cmegalodon** [comment]
Yes, it shows me the amps off of the main CTs

![image](https://github.com/magico13/ha-emporia-vue/assets/97914057/591891c2-5f93-49ba-9b0d-e07ab9aab7a6)

Here's the two views in the app to see. Ignore L1+L2, those are additional CTs.

**Comment by DabblerIOT** [comment]
Ill be damned it shares a/b data that way let me pull it on api.  will
check after watching golf :)

Valid use case / way to get data imho.

Someone else asked something similar awhile ago..

**Comment by cmegalodon** [comment]
It's the only part of the app that shows it, you can't graph the amps. If the Emporia API can exposes it, I think it would be a great addition.

Thank you for being so responsive :)

**Comment by DabblerIOT** [comment]
@magico13 

Using units of AmpHours gives the following output.   This data is useful for certain applications to see the load on A and B power feeds.  (Both sides of the center tap transformer in US).

I believe this is the only way you can retrieve the readings of both CTs on the mains.

{
    "deviceListUsages": {
        "instant": "2024-04-12T14:28:35Z",
        "scale": "1MIN",
        "devices": [
            {
                "deviceGid": 296617,
                "channelUsages": [
                    {
                        "name": "Main",
                        "usage": 0.14615494404015716,
                        "deviceGid": 296617,
                        "channelNum": "1,2,3",
                        "percentage": 100,
                        "nestedDevices": []
                    },
                    {
                        "name": "Mains_A",
                        "usage": 0.11057835136519538,
                        "deviceGid": 296617,
                        "channelNum": "Mains_A",
                        "percentage": 75.65830365260388,
                        "nestedDevices": []
                    },
                    {
                        "name": "Mains_B",
                        "usage": 0.0355765926749618,
                        "deviceGid": 296617,
                        "channelNum": "Mains_B",
                        "percentage": 24.341696347396134,
                        "nestedDevices": []
                    },
                    {
                        "name": "Guest Bedroom",
                        "usage": 0.014455516006107684,
                        "deviceGid": 296617,
                        "channelNum": "1",
                        "percentage": 9.890541918401285,
                        "nestedDevices": []
                    },
                    {
                        "name": "Great Room",
                        "usage": 0.011287512359795747,
                        "deviceGid": 296617,
                        "channelNum": "2",
                        "percentage": 7.722976758620234,
                        "nestedDevices": []
                    },
                    {
                        "name": "House GFI / Chest Freezer",
                        "usage": 0.04124339225998631,
                        "deviceGid": 296617,
                        "channelNum": "3",
                        "percentage": 28.218951148620995,
                        "nestedDevices": []
                    },
                    {
                        "name": "Bathroom Plugs",
                        "usage": 0,
                        "deviceGid": 296617,
                        "channelNum": "4",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Kitchen Fridge",
                        "usage": 0.03639037370240247,
                        "deviceGid": 296617,
                        "channelNum": "5",
                        "percentage": 24.898489709936836,
                        "nestedDevices": []
                    },
                    {
                        "name": "Wet Bar",
                        "usage": 0.0001332101970911026,
                        "deviceGid": 296617,
                        "channelNum": "6",
                        "percentage": 0.09114313440844128,
                        "nestedDevices": []
                    },
                    {
                        "name": "AC2",
                        "usage": 0,
                        "deviceGid": 296617,
                        "channelNum": "7",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Dryer Floor 1",
                        "usage": 0,
                        "deviceGid": 296617,
                        "channelNum": "8",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Pool House A",
                        "usage": 0.00399504323579647,
                        "deviceGid": 296617,
                        "channelNum": "9",
                        "percentage": 2.733430101891594,
                        "nestedDevices": []
                    },
                    {
                        "name": "Pool House B",
                        "usage": 0,
                        "deviceGid": 296617,
                        "channelNum": "10",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Pool A",
                        "usage": 0.015180280283645346,
                        "deviceGid": 296617,
                        "channelNum": "11",
                        "percentage": 10.386429541154934,
                        "nestedDevices": []
                    },
                    {
                        "name": "Pool B",
                        "usage": 0.014056516451968086,
                        "deviceGid": 296617,
                        "channelNum": "12",
                        "percentage": 9.61754427418203,
                        "nestedDevices": []
                    },
                    {
                        "name": "Garage Dan",
                        "usage": 0.005479926512749107,
                        "deviceGid": 296617,
                        "channelNum": "14",
                        "percentage": 3.7493952385513936,
                        "nestedDevices": []
                    },
                    {
                        "name": "Hall Lights / Upright Freezer",
                        "usage": 0.007428204584452841,
                        "deviceGid": 296617,
                        "channelNum": "15",
                        "percentage": 5.082417589932425,
                        "nestedDevices": []
                    },
                    {
                        "name": "Kitchen Lights",
                        "usage": 0.007385274235849028,
                        "deviceGid": 296617,
                        "channelNum": "16",
                        "percentage": 5.053044414166289,
                        "nestedDevices": []
                    }
                ]
            },
            {
                "deviceGid": 296746,
                "channelUsages": [
                    {
                        "name": "Main",
                        "usage": 0.1485868194323999,
                        "deviceGid": 296746,
                        "channelNum": "1,2,3",
                        "percentage": 100,
                        "nestedDevices": []
                    },
                    {
                        "name": "Mains_A",
                        "usage": 0.028955224677368444,
                        "deviceGid": 296746,
                        "channelNum": "Mains_A",
                        "percentage": 19.48707482129108,
                        "nestedDevices": []
                    },
                    {
                        "name": "Mains_B",
                        "usage": 0.11963159475503145,
                        "deviceGid": 296746,
                        "channelNum": "Mains_B",
                        "percentage": 80.51292517870891,
                        "nestedDevices": []
                    },
                    {
                        "name": "Dryer 2",
                        "usage": 0,
                        "deviceGid": 296746,
                        "channelNum": "1",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Dining Room",
                        "usage": 0,
                        "deviceGid": 296746,
                        "channelNum": "2",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Bathroom Lights",
                        "usage": 0.019951856944296093,
                        "deviceGid": 296746,
                        "channelNum": "3",
                        "percentage": 13.427743470458536,
                        "nestedDevices": []
                    },
                    {
                        "name": "Foyer",
                        "usage": 0.0028498143592366465,
                        "deviceGid": 296746,
                        "channelNum": "4",
                        "percentage": 1.9179455957957157,
                        "nestedDevices": []
                    },
                    {
                        "name": "Basement Office",
                        "usage": 0.02065263106646361,
                        "deviceGid": 296746,
                        "channelNum": "5",
                        "percentage": 13.899369503537692,
                        "nestedDevices": []
                    },
                    {
                        "name": "Up Office / Hall",
                        "usage": 0,
                        "deviceGid": 296746,
                        "channelNum": "6",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Indoor Unit Up",
                        "usage": 0.006122618408114821,
                        "deviceGid": 296746,
                        "channelNum": "7",
                        "percentage": 4.120566300229831,
                        "nestedDevices": []
                    },
                    {
                        "name": "Indoor Unit Down",
                        "usage": 0.0060354950968865995,
                        "deviceGid": 296746,
                        "channelNum": "8",
                        "percentage": 4.061931684076776,
                        "nestedDevices": []
                    },
                    {
                        "name": "Well Pump",
                        "usage": 0.011461127565966714,
                        "deviceGid": 296746,
                        "channelNum": "9",
                        "percentage": 7.7134214257685185,
                        "nestedDevices": []
                    },
                    {
                        "name": "Garage Dawn",
                        "usage": 0,
                        "deviceGid": 296746,
                        "channelNum": "10",
                        "percentage": 0,
                        "nestedDevices": []
                    },
                    {
                        "name": "Outside Plugs?",
                        "usage": 0.0025947578560422966,
                        "deviceGid": 296746,
                        "channelNum": "11",
                        "percentage": 1.7462907315428409,
                        "nestedDevices": []
                    },
                    {
                        "name": "Outside Plugs Back",
                        "usage": 0.006864428728818894,
                        "deviceGid": 296746,
                        "channelNum": "12",
                        "percentage": 4.619809990577186,
                        "nestedDevices": []
                    },
                    {
                        "name": "Nado Rooms",
                        "usage": 0.01507547996441523,
                        "deviceGid": 296746,
                        "channelNum": "13",
                        "percentage": 10.145906630213505,
                        "nestedDevices": []
                    },
                    {
                        "name": "Basement ManCave Plugs",
                        "usage": 0.050229082500493086,
                        "deviceGid": 296746,
                        "channelNum": "14",
                        "percentage": 33.80453440780794,
                        "nestedDevices": []
                    },
                    {
                        "name": "Basement Lights",
                        "usage": 0.005536746012943763,
                        "deviceGid": 296746,
                        "channelNum": "15",
                        "percentage": 3.726269957250633,
                        "nestedDevices": []
                    },
                    {
                        "name": "AC 1",
                        "usage": 0,
                        "deviceGid": 296746,
                        "channelNum": "16",
                        "percentage": 0,
                        "nestedDevices": []
                    }
                ]
            }
        ],
        "energyUnit": "AmpHours"
    }
}

**Comment by magico13** [comment]
Yeah it's definitely possible, there's another issue out there I think for this request. On my list of things to do but haven't gotten to it yet. Really this is only helpful for the Vue device, not channels or plugs, so to keep noise down I'd probably only add the different phases to whatever vues are on the account.

**Comment by DabblerIOT** [comment]
Agree, it's a shame that the getdeviceusages with kw doesn't give watts of these two cts independantly.   Not sure if every user would want/need this.   Think it is more of a data point for those with generator backup.   Seems like a lot of requests and data to basically get 2 data points.

**Comment by DabblerIOT** [comment]
Its just these two data points, its a shame they don't come through with the power request..  Been looking at code, annoying from that side of it..


{
"name": "Mains_A",
"usage": 0.11057835136519538,
"deviceGid": 296617,
"channelNum": "Mains_A",
"percentage": 75.65830365260388,
"nestedDevices": []
},
{
"name": "Mains_B",
"usage": 0.0355765926749618,
"deviceGid": 296617,
"channelNum": "Mains_B",
"percentage": 24.341696347396134,
"nestedDevices": []
},

**Comment by andreitoma-github** [comment]
Just want to let you know that I added a VUE Gen3 and this HA integration worked perfectly.
I never had a previous generations of VUE Energy Monitor, but this Gen3 works excellent with Home Assistant.
Seeing the Watts, Volts and Amps per each phase will be a great addition !

**Comment by DabblerIOT** [comment]
I didn't forget about you.. still thinking about you... its just somewhat annoying this doesn't come through on the kw feed, as we are under requests per day request from emporia for the code here.

**Comment by DabblerIOT** [comment]
> Just want to let you know that I added a VUE Gen3 and this HA integration worked perfectly. I never had a previous generations of VUE Energy Monitor, but this Gen3 works excellent with Home Assistant. Seeing the Watts, Volts and Amps per each phase will be a great addition !

its more the amps for phase a and b on the main cts that this topic is about.   The rest if you really care you can math with template yourself in ha if you really care...    volts doesn't come across at all in the api btw.

:)

**Comment by cmegalodon** [comment]
Thanks for the updates! I really appreciate everything you are doing for the HA community!

**Comment by Arson114** [comment]
Having amps would be amazing! I use those stats regularly on the main and sub CTs. Thanks!

**Comment by BenJewell** [comment]
Adding this feature might also be a good time to implement:
https://github.com/magico13/ha-emporia-vue/issues/246

**Comment by bverkron** [comment]
I have a major need for knowing the amps from the Emporia Vue 2 that I have. Attempting to calculate it with a template sensor is problematic because of things like power factor (apparently) which varies based on load, it seems. I'm having a hard time getting the calculated amps in HA to match what Emporia shows. Having the amps data directly would save a lot of hassle and inaccuracies.
